.github/pull_request_template.md

```diff
@@ -0,0 +1,25 @@
+## Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Fixes # (issue)
+
+## Related Issues:
+- # (issue)
+
+## Type of change
+
+Please delete options that are not relevant.
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## Checklist:
```

Comment:
This checklist also seems overkill to me. Although the bullet point about the documentation is relevant. It would make more sense to put it into the issue also. Same for tests. The last bullet point should also not be there. We should not be allowed to merge a PR if the tests are failing. Making the branch protected should handle that.

Reply:
Removed the pass tests checklist.
I think other points are relevant. no?

Reply:
ppl tend to forget. I mean I saw so far PRs that missing required migrations, PRs that missing documentations and PRs that not follows the commits conventions. this why the checklist make sense to me.

Reply:
I’m not sure that forcing these points into the issue is more reasonable.
Anyone can open issues, even non-technical users. Issues are usually about something that doesn’t work or about suggestion, and the changes required to sort this out could be unclear from the issue alone or known to the user. It could be that the user is not able to understand how to make it work, so the developer adds documentation. It could be an actual bug that requires a fix or something that is not even supported and if accepted requires a new feature. So, an issue usually is about a problem not about the changes requires to fix this problem and a PR should include the details on how we fixed this problem.

A good pull request is one that you understand the point of, clarifies the intentions of the code changes, summarizes the code well and identifies some of the thought process involved in writing the code.

Also, a great pull request answers some questions before they are asked.

Reply:
I’m not sure that forcing these points into the issue is more reasonable.
Anyone can open issues, even non-technical users. Issues are usually about something that doesn’t work or about suggestion, and the changes required to sort this out could be unclear from the issue alone or known to the user. It could be that the user is not able to understand how to make it work, so the developer adds documentation. It could be an actual bug that requires a fix or something that is not even supported and if accepted requires a new feature. So, an issue usually is about a problem not about the changes requires to fix this problem and a PR should include the details on how we fixed this problem.

A good pull request is one that you understand the point of, clarifies the intentions of the code changes, summarizes the code well and identifies some of the thought process involved in writing the code.

Also, a great pull request answers some questions before they are asked.

Reply:
Everyone can indeed open issues. Maybe you're right and maybe it should go in the PR. Though I would remove some bullet points:
- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] All new and existing tests passed.
For these reasons:
- Your code should always follow the code style of the project
- Everyone should have read that document before working on tfchain and once you have read it you will have to check that box for every PR which is a bit of a hassle
- The tests should always pass before merging and we enforce that by not being able to merge the PR

Also these two bullet points go hand in hand, I would only put one that says both:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Maybe rewrite to:
- [ ] My changes requires a change to the documentation and I have updated it

Reply:
Okay. Can you check now?